pkg/crds/controllers/batch/batchjob_controller.go

```diff
@@ -231,6 +231,12 @@ func (r *BatchJobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		log.Info("failed to enqueue payload")
 	case batch.EnqueuingDone:
 		if !workerJobExists {
+			log.V(1).Info("creating worker configmap")
+			if err = r.createWorkerConfigMap(ctx, batchJob); err != nil {
+				log.Error(err, "failed to create worker configmap")
+				return ctrl.Result{}, err
+			}
+
```

Comment:
So that would mean that we need to check that it hasn't been created before, and if not, then just skip. I'm seeing that this isn't done for the worker job, so why would this be required for this? Or put it differently, maybe the worker job needs the same check as well?

Had a longer job running and I don't recall seeing any errors in the controller manager.

Reply:
it is done for the worker job

Reply:
the happy path will still work, it's only a problem if something fails